tests/run-make/print-in-stable-or-unstable-channel/rmake.rs

```diff
@@ -0,0 +1,96 @@
+use run_make_support::rustc;
+
+fn main() {
+    let unstable_print_requests = [
+        "all-target-specs-json",
+        "check-cfg",
+        "crate-root-lint-levels",
+        "supported-crate-types",
+        "target-spec-json",
+    ];
+    let stable_print_requests = [
+        "calling-conventions",
+        "cfg",
+        "code-models",
+        "crate-name",
+        "deployment-target",
+        "file-names",
+        "host-tuple",
+        "link-args",
+        "native-static-libs",
+        "relocation-models",
+        "split-debuginfo",
+        "stack-protector-strategies",
+        "sysroot",
+        "target-cpus",
+        "target-features",
+        "target-libdir",
+        "target-list",
+        "tls-models",
+    ];
+
+    // For issue-138698
+    // Failed. In stable channel, only stable args are printed when an invalid arg is provided
+    let output = rustc()
+        .env("RUSTC_BOOTSTRAP", "-1")
+        .cfg("force_stable")
+        .print("xxx")
+        .run_fail()
+        .stderr_utf8();
+    for request in unstable_print_requests {
+        assert!(!output.contains(request));
+    }
+
+    // For issue-138698
+    // Failed. In unstable channel, all args are print when an invalid arg is provided
+    let output = rustc().print("xxx").run_fail().stderr_utf8();
+    for request in unstable_print_requests {
+        assert!(output.contains(request));
+    }
+
+    // Other tests, for future improvements
+    // Failed. In unstable channel, only stable args are print when a valid arg is provided
+    let output = rustc()
+        .print(unstable_print_requests[0])
+        .print(stable_print_requests[0])
+        .run_fail()
+        .stderr_utf8();
+    assert!(output.contains(
+        "error: the `-Z unstable-options` flag must also be \
+passed to enable the `all-target-specs-json` print option"
+    ));
+
+    // Ok. In unstable channel, only unstable args are print
+    // when a valid arg is provided with `-Z unstable-options`
+    rustc()
+        .arg("-Z")
+        .arg("unstable-options")
+        .print(unstable_print_requests[0])
+        .print(stable_print_requests[0])
+        .run();
+
+    // Failed. In stable channel, unstable args are reminded to add `-Z unstable-options`
+    let output = rustc()
+        .env("RUSTC_BOOTSTRAP", "-1")
+        .cfg("force_stable")
+        .print(unstable_print_requests[0])
+        .run_fail()
+        .stderr_utf8();
+    assert!(output.contains(
+        "error: the `-Z unstable-options` flag must also be passed to \
+enable the `all-target-specs-json` print option"
+    ));
```

Comment:
Suggestion: remove this, this is redundant with `tests/ui/print-request/stability.rs`